Modules/Sources/WordPressUI/Views/Users/ViewModel/UserDeleteViewModel.swift

```diff
@@ -0,0 +1,104 @@
+import SwiftUI
+
+@MainActor
+public class UserDeleteViewModel: ObservableObject {
+
+    @Published
+    var isFetchingOtherUsers: Bool = false
+
+    @Published
+    var isDeletingUser: Bool = false
+
+    @Published
+    var error: Error? = nil
+
+    @Published
+    var otherUserId: Int32 = 0
+
+    @Published
+    var otherUsers: [DisplayUser] = []
+
+    @Published
+    var deleteButtonIsDisabled: Bool = true
+
+    private let userProvider: UserDataProvider
+    private let actionDispatcher: UserManagementActionDispatcher
+    let user: DisplayUser
+
+    init(user: DisplayUser, userProvider: UserDataProvider, actionDispatcher: UserManagementActionDispatcher) {
+        self.user = user
+        self.userProvider = userProvider
+        self.actionDispatcher = actionDispatcher
+    }
+
+    func fetchOtherUsers() async {
+        withAnimation {
```

Comment:
(nit) It works, but I'd suggest looking into [animation(_:value:)](https://developer.apple.com/documentation/swiftui/view/animation(_:value:)) for specifying animations in the view layer.

Reply:
+ same comments are about `private(set)` and `@MainActor` in task apply here.